compiler/rustc_resolve/src/def_collector.rs

```diff
@@ -285,19 +285,36 @@ impl<'a, 'b> visit::Visitor<'a> for DefCollector<'a, 'b> {
             TyKind::MacCall(..) => self.visit_macro_invoc(ty.id),
             TyKind::ImplTrait(node_id, _) => {
                 let parent_def = match self.impl_trait_context {
-                    ImplTraitContext::Universal(item_def) => self.resolver.create_def(
-                        item_def,
-                        node_id,
-                        DefPathData::ImplTrait,
-                        self.expansion.to_expn_id(),
-                        ty.span,
-                    ),
+                    ImplTraitContext::Universal(item_def) => {
+                        //| ImplTraitContext::UniversalInDyn(item_def) => {
+                        let def_id = self.resolver.create_def(
+                            item_def,
+                            node_id,
+                            DefPathData::ImplTrait,
+                            self.expansion.to_expn_id(),
+                            ty.span,
+                        );
+                        self.resolver
+                            .impl_trait_context
+                            .insert(def_id, ImplTraitContext::Universal(item_def));
```

Comment:
```suggestion
                            .insert(def_id, self.impl_trait_context);
```
We have not exited the trait object.

Reply:
My bad, your code is right.